doc: Maven html documentation generating library (closes #27)

edgarwolff opened the pull request:
Added in pom.xml the dependency allowing to produce html documentation

Documentation can be produced by running mvn javadoc:javadoc and is available in /target/reports/apidocs/CIServer.html

edgarwolff committed:
doc: Maven html documentation generating library

Added in pom.xml the dependency allowing to produce html documentation

WatermelonGodz commented on pom.xml:
```diff
@@ -54,5 +54,32 @@
+                <artifactId>maven-javadoc-plugin</artifactId>
+                <version>3.11.2</version>
+                <configuration>
+                    <stylesheetfile>${basedir}/src/main/javadoc/stylesheet.css</stylesheetfile>
```
I can not find the folder nor a css file in this location

edgarwolff replied:
Indeed, it is a default value set there but if no CSS is found default will be used, but i'll delete this in a further commit, I'm also writing javadoc for ALL the public entities (not only methods but also constructors classes (yes javadoc for the whole class as one)

edgarwolff committed:
doc: Update doc gen library configuration

Removed unnecessary configuration option
Changed report output location to avoid being in ignored folder

edgarwolff committed:
doc: Documenting public entities

Documented all public entities already defined in the current state of the repository

WatermelonGodz commented on src/main/java/CIServer.java:
```diff
@@ -12,9 +12,20 @@
 public class CIServer extends AbstractHandler {
 
+    /**
+     * Empty default constructor, no resource nor parameter need for instantiation
```
Spelling

KristinRosen commented on src/main/java/server_communication/Email.java:
Looks good! :) I'm just gonna change "Send" to "send" for naming convention reasons

KristinRosen committed:
Refact: renamed 'Send()' to 'send()'

beginner003 reviewed (dismissed):
Looks good to me

edgarwolff committed:
test: Empty commit for CI trigger

beginner003 committed:
doc: Update README.md (draft)

This update is a draft.

beginner003 committed:
Update README.md (2)

- included the title and a cover image

beginner003 committed:
Uploaded cover image for README.md

beginner003 committed:
Update README.md (3)

- updated the link to the cover image

edgarwolff committed:
test: Emtpy to trigger CI

beginner003 committed:
Update README.md (4)

- Added dependencies

beginner003 committed:
Merge branch 'edgar/doc/browsable-javadoc' of https://github.com/DD2480-Group-27/assignment2-CI into edgar/doc/browsable-javadoc

edgarwolff committed:
test: Emtpy to trigger CI

edgarwolff committed:
doc: Adding generated documentation

Added the html documentation

blimpan committed:
Added to system reqs, instructions for running and contributions

blimpan committed:
Added info about API key and did some styling

edgarwolff committed:
Merge branch 'main' into edgar/doc/browsable-javadoc

# Conflicts:
#	README.md
#	src/main/java/CIServer.java
#	src/main/java/server_communication/Email.java

edgarwolff committed:
refact fixing compilation error after merge

edgarwolff committed:
Regenerating browsable javadoc

blimpan reviewed (approved):
LGTM!
